### PR TITLE
Added DASH Blockchain scraper

### PR DESCRIPTION
**Pull request by ishanjain28**

### CODE REVIEW

File: blockchain-scrapers/cmd/dash/dash.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/diadata-org/api-golang/dia"
+)
+
+const (
+	HOST     = "localhost"
```
Comment (frranck) [inline blockchain-scrapers/cmd/dash/dash.go]: Hello, I think you forgot to add a dash node in docker-compose and then this should point to it and not localhost.

Comment (ishanjain28) [reply blockchain-scrapers/cmd/dash/dash.go]: @frranck Okay, I updated the host.

**Comment by frranck** [comment]
Shouldn't there be two services in the docker compose ? one for the scrapper and one for the dask node ? then the scrapper would refers to the other one to do the RPC call.

**Comment by ishanjain28** [comment]
@frranck Yes, I am sorry. I forgot to add dashd service. 

I have added it now

**Comment by frranck** [comment]
I think we're almost there, but it's complaining about the rpc password:

> 
> 2018-09-03 14:16:16 ThreadRPCServer incorrect password attempt from 10.0.0.180:54632
> 2018-09-03 14:16:21 CMasternodeSync::ProcessTick -- nTick 91 nRequestedMasternodeAssets 0 nRequestedMasternodeAttempt 0 nSyncProgress -0.250000
> 2018-09-03 14:16:25 ThreadRPCServer incorrect password attempt from 10.0.0.181:43188
> 2018-09-03 14:16:27 CMasternodeSync::ProcessTick -- nTick 97 nRequestedMasternodeAssets 0 nRequestedMasternodeAttempt 0 nSyncProgress -0.250000
> 2018-09-03 14:16:33 CMasternodeSync::ProcessTick -- nTick 103 nRequestedMasternodeAssets 0 nRequestedMasternodeAttempt 0 nSyncProgress -0.250000
> 2018-09-03 14:16:33 ThreadRPCServer incorrect password attempt from 10.0.0.182:57272
> 2018-09-03 14:16:39 CMasternodeSync::ProcessTick -- nTick 109 nRequestedMasternodeAssets 0 nRequestedMasternodeAttempt 0 nSyncProgress -0.250000
> 2018-09-03 14:16:42 ThreadRPCServer incorrect password attempt from 10.0.0.183:56338
> 2018-09-03 14:16:45 CMasternodeSync::ProcessTick -- nTick 115 nRequestedMasternodeAssets 0 nRequestedMasternodeAttempt 0 nSyncProgress -0.250000
>

**Comment by ishanjain28** [comment]
@frranck Fixed that bug by adding a `command` field in docker-compose configuration.

**Comment by frranck** [comment]
there is one  more problem:
you're sending old values, you should check that we are at the tip of the blockchain, 
use a system like we use in BTC and LTC
https://github.com/diadata-org/api-golang/blob/master/blockchain-scrapers/scrapers/blockchain-scrapers.go#L65

also the symbol is DSH and not DASH

**Comment by ishanjain28** [comment]
Hi 

If I use the `blockchain_scraper` package to fetch dash's circulating supply, I get 

```
2018/09/06 04:25:14 Loaded secret in /home/ishan/config/secrets/api_diadata.json
2018/09/06 04:25:15 GetBlockchainInfo: {0 false 1390095618}
2018/09/06 04:25:15 ElapsedTime block: 40581h14m57.856619474s 50 0
2018/09/06 04:25:25 GetBlockchainInfo: {0 false 1390095618}
2018/09/06 04:25:25 ElapsedTime block: 40581h15m7.857821673s 50 0
```

My node was running for 20 minutes. The program was similar to that of btc/ltc and it was not submitting anything to diadata client.


I had asked DASH developers for a way to figure out circulating supply of dash in a network and my current implementation is based on what they told me.

discord chat 

If my current implementation is incorrect, Should I replace it with the code in btc/ltc? Is there anything I'll have to change in that code?

**Comment by frranck** [comment]
Yes please do something like for bitcoin or litecoin, use this package
https://github.com/diadata-org/api-golang/blob/master/blockchain-scrapers/scrapers/blockchain-scrapers.go it wraps the whole "is it the tip?" thing and uses "github.com/diadata-org/go-bitcoind" to communicate with the RPC.
then fine tune the correct number for the tip detection: 
https://github.com/diadata-org/api-golang/blob/master/blockchain-scrapers/cmd/btc/btc.go#L14

**Comment by frranck** [comment]
Hi Ishan, any update on this ?

**Comment by frranck** [comment]
@ishanjain28 it's just a small refactoring, do you want to finish it ? If I don't hear from you we will have to close and reopen the bounty.

**Comment by farhaanbukhsh** [comment]
hey @frranck should I finish this PR?

**Pull request closed without merging**